pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1598,4 +1598,202 @@ public void testNoParametersNPE() throws SQLException {
       // ignore
     }
   }
+
+  @Test
+  public void testCountOfInsertForMutableStatementWithSetBefore() throws SQLException {
+    try (PreparedStatement pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?)")) {
+      pstmt.setInt(1, 1);
+      int count = pstmt.executeUpdate();
+      assertEquals("Expected update count to be 1", 1, count);
+    }
+  }
+
+  @Test
+  public void testCountOfBatchInsertForMutableStatementWithSetBefore() throws SQLException {
+    try (PreparedStatement pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?)")) {
+      pstmt.setInt(1, 1);
+      pstmt.addBatch();
+      pstmt.setInt(1, 2);
+      pstmt.addBatch();
+      int[] counts = pstmt.executeBatch();
+      assertEquals("Expected update count to be 2", 2, counts.length);
+      assertEquals("Expected update count for first batch to be 1", 1, counts[0]);
+      assertEquals("Expected update count for second batch to be 1", 1, counts[1]);
+    }
+  }
+
+  @Test
+  public void testCountOfInsertForMutableStatementWithSetSurrounding() throws SQLException {
+    try (PreparedStatement  pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?);SET search_path = 'public';")) {
+      pstmt.setInt(1, 1);
+      int count = pstmt.executeUpdate();
+      assertEquals("Expected update count to be 1", 1, count);
```

Comment:
`Expected update count to be 1` does not add much value.
JUnit will print "expected 1 got 4 anyway".

The message should clarify the reason why a certain value was expected.
In this case, `assertEquals("pstmt.executeUpdate(): inserting one row", 1, count);` would be better.

A sample failure message would look like `pstmt.executeUpdate(): inserting one row, expected 1 got 4`